### PR TITLE
Update to Gradle 7 + Fixes to avoid crash with multiple cameras or no frame size selected

### PR DESCRIPTION
**Pull request by ChainReaction31**
Added a try catch when initializing the camera that should keep a crash from occurring.
Also added a warning that appears when an incorrect preset is selected in video settings, prompting a user to check them

### CODE REVIEW

File: sensorhub-driver-android/src/main/java/org/sensorhub/impl/sensor/android/AndroidSensorsDriver.java [hunk 1]
```diff
@@ -182,65 +191,93 @@ protected void createCameraOutputs(Context androidContext) throws SensorExceptio
         }
         else*/
         {
+//            for(int cameraId = 0; cameraId < android.hardware.CameraX.getNumberOfCameras(); cameraId++)
```
Comment (alexrobin) [inline sensorhub-driver-android/src/main/java/org/sensorhub/impl/sensor/android/AndroidSensorsDriver.java:194]: I think we should leave this out from the PR

File: sensorhub-driver-android/AndroidManifest.xml [hunk 2]
```diff
@@ -2,4 +2,6 @@
     package="org.sensorhub.impl.sensor.android"
     android:versionCode="1"
     android:versionName="1.0" >
+
+    <uses-permission android:name="android.permission.CAMERA" />
```
Comment (alexrobin) [inline sensorhub-driver-android/AndroidManifest.xml:6]: I don't think we need this here since we have it on the app manifest. Can you try without?

Comment (ChainReaction31) [reply sensorhub-driver-android/AndroidManifest.xml]: Pushed up a fix for both.

File: sensorhub-driver-android/src/main/java/org/sensorhub/impl/sensor/android/video/AndroidCameraOutput.java [hunk 3]
```diff
@@ -209,16 +194,13 @@ public void run()
...
-            try
-            {
+        if (camera != null) {
+            try {
                 Parameters camParams = camera.getParameters();
 
                 // set video capture and encodign options
                 VideoEncoderConfig videoConfig = parentSensor.getConfiguration().videoConfig;
                 frameRate = videoConfig.frameRate;
-                selectedPreset = videoConfig.selectedPreset;
```
Comment (alexrobin) [inline sensorhub-driver-android/src/main/java/org/sensorhub/impl/sensor/android/video/AndroidCameraOutput.java:221]: Sorry I didn't look at this file before.
Why was this line removed?

Comment (ChainReaction31) [reply sensorhub-driver-android/src/main/java/org/sensorhub/impl/sensor/android/video/AndroidCameraOutput.java]: Should not have been. I was trying something there and it got caught up in taking that code back out. The fact that selected preset was defined elsewhere happened to keep me and the compiler from catching it.

Pushed a commit with the line added back.

File: sensorhub-driver-android/src/main/java/org/sensorhub/impl/sensor/android/AndroidSensorsDriver.java [hunk 4]
```diff
@@ -7,18 +7,23 @@
...
 
+import static android.content.Context.CAMERA_SERVICE;
+
 import java.util.ArrayList;
 import java.util.List;
 import javax.xml.namespace.QName;
 
 import android.graphics.SurfaceTexture;
+import android.hardware.camera2.CameraAccessException;
+import android.hardware.camera2.CameraManager;
```
Comment (alexrobin) [inline sensorhub-driver-android/src/main/java/org/sensorhub/impl/sensor/android/AndroidSensorsDriver.java:25]: Can you remove these camera2 imports? I think they are unused

File: sensorhub-driver-android/src/main/java/org/sensorhub/impl/sensor/android/AndroidSensorsDriver.java [hunk 5]
```diff
@@ -51,63 +56,66 @@
 import android.os.Build;
 import android.provider.Settings.Secure;
 
+//import androidx.camera
+import android.hardware.camera2.CameraManager;
```
Comment (alexrobin) [inline sensorhub-driver-android/src/main/java/org/sensorhub/impl/sensor/android/AndroidSensorsDriver.java:60]: Can you remove these imports. I think they are unused

**Review by alexrobin — COMMENTED** [review]
Can you completely revert changes to AndroidCameraOutput.java. It looks like changes are just automatic formatting from the IDE.

File: sensorhub-driver-android/src/main/java/org/sensorhub/impl/sensor/android/AndroidSensorsDriver.java (hunk 5 above)
Comment (ChainReaction31) [reply sensorhub-driver-android/src/main/java/org/sensorhub/impl/sensor/android/AndroidSensorsDriver.java]: pushed commit to restore previous version of the file and one to optimize the commits

File: sensorhub-driver-android/src/main/java/org/sensorhub/impl/sensor/android/video/AndroidCameraOutput.java [hunk 6]
```diff
@@ -7,9 +7,9 @@
 Software distributed under the License is distributed on an "AS IS" basis,
 WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License
 for the specific language governing rights and limitations under the License.
-
```
Comment (ChainReaction31) [inline sensorhub-driver-android/src/main/java/org/sensorhub/impl/sensor/android/video/AndroidCameraOutput.java:10]: I checked the file out from the last commit you made. I think the IDE still just automatically stripped some of the whitespace characters out.

Comment (alexrobin) [reply sensorhub-driver-android/src/main/java/org/sensorhub/impl/sensor/android/video/AndroidCameraOutput.java:10]: I think you should disable auto-formatting in your IDE when you modify existing code (especially if it has different formatting rules). I used to have it enabled too but I realized it makes it really hard to look at commit diffs because the important changes are lost in the middle of zillions of formatting changes.

You can leave it enabled when you create your own code but then disable it when you modify existing code.

**Comment by ChainReaction31** [comment]
@alexrobin AndroidCameraOutput doesn't show in the changed files now, so it should be correctly reverted. Still don't know what caused the diff because I don't have autoformat on.

**Review by alexrobin — APPROVED** [review]
